xbmc/epg/EpgContainer.cpp

```diff
@@ -245,16 +245,17 @@ bool CEpgContainer::PersistAll(void)
 {
   bool bReturn(true);
   m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
+  EPGMAP copy = m_epgs;
   m_critSection.unlock();
 
-  for (EPGMAP_CITR it = copy.begin(); it != copy.end() && !m_bStop; it++)
+  for (const auto &epgEntry : copy)
   {
-    CEpg *epg = it->second;
+    CEpg *epg = epgEntry.second;
     if (epg && epg->NeedsSave())
-    {
       bReturn &= epg->Persist();
-    }
+
+    if (m_bStop)
```

Comment:
Fancy new notation has no advantages here with the condition moved out of the for statement. I'd keep the old notation here, so the break condition is defined in the for statement. and while you're at it, move the `CEpg* epg` declaration a few lines up so we don't create a new one in each iteration ;-)